tutorials/composite_mtbo.ipynb

```diff
@@ -0,0 +1,740 @@
+    "\n",
+    "For greater than $10$ or so tasks, this would have been pretty much infeasible to sample the posterior over all tasks jointly, except that [Maddox et al, '21](https://arxiv.org/abs/2106.12997) have devised an efficient method for exploiting the structure in the posterior distribution of the MTGP, enabling efficient MC based optimization of objectives using MTGPs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "problem = ContextualHartmann6(noise_std = 0.001, negate=True).to(**tkwargs)\n",
+    "\n",
+    "weights = torch.randn(20, **tkwargs)\n",
+    "def callable_func(samples, X=None):\n",
+    "    res = -torch.cos((samples**2) + samples * weights)\n",
+    "    return res.sum(dim=-1)\n",
+    "objective = GenericMCObjective(callable_func)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bounds = problem.bounds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define helper functions\n",
+    "\n",
+    "Define helper functions used for optimizing the acquisition function and for constructing the batch expected improvement acquisition, which we optimize for both the batch GP and MTGP."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def optimize_acqf_and_get_candidate(acq_func, bounds, batch_size):\n",
+    "    \"\"\"Optimizes the acquisition function, and returns a new candidate and a noisy observation.\"\"\"\n",
+    "    # optimize\n",
+    "    candidates, _ = optimize_acqf(\n",
+    "        acq_function=acq_func,\n",
+    "        bounds=bounds,\n",
+    "        q=batch_size,\n",
+    "        num_restarts=10,\n",
+    "        raw_samples=512,  # used for intialization heuristic\n",
+    "        options={\"batch_limit\": 5, \"maxiter\": 200, \"init_batch_limit\": 5},\n",
+    "    )\n",
+    "    # observe new values \n",
+    "    new_x = candidates.detach()\n",
+    "    return new_x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def construct_acqf(model, objective, num_samples, best_f):\n",
+    "    sampler = IIDNormalSampler(num_samples=num_samples)\n",
+    "    qEI = qExpectedImprovement(\n",
+    "        model=model, \n",
+    "        best_f=best_f,\n",
+    "        sampler=sampler, \n",
+    "        objective=objective,\n",
+    "    )\n",
+    "    return qEI"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_init = 20\n",
```

Comment:
suggest moving this into the SMOKE_TEST as well and choosing a smaller number to speed things up even more